sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLQueryPlanner.java

```diff
@@ -89,7 +89,9 @@
 public class ZetaSQLQueryPlanner implements QueryPlanner {
   // TODO(BEAM-11747) Re-enable BeamJavaUdfCalcRule by default when it is safe to do so.
   public static final Collection<RelOptRule> DEFAULT_CALC =
-      ImmutableList.<RelOptRule>builder().add(BeamZetaSqlCalcRule.INSTANCE).build();
+      ImmutableList.<RelOptRule>builder()
+          .add(BeamZetaSqlCalcRule.INSTANCE, BeamZetaSqlCalcMergeRule.INSTANCE)
```

Comment:
Will it make sense to create two RuleSets? One for each dialect?

Reply:
Maybe. The downside to the current approach is the modifying rule sets is slightly more complex; the upside is that we don't have to worry about someone adding a new rule to Calcite and forget about adding it to ZetaSQL.